blockchain/chainio.go

```diff
@@ -1770,9 +1777,16 @@ func (b *BlockChain) initChainState(interrupt <-chan struct{}) error {
 		block := utilBlock.MsgBlock()
 		blockSize := uint64(block.SerializeSize())
 		numTxns := uint64(len(block.Transactions))
+
+		// Calcultate the next stake difficulty.
+		nextStakeDiff, err := b.CalcNextRequiredStakeDifficulty()
```

Comment:
You should use `b.calcNextRequiredStakeDifficulty(tip)` instead to be sure you're calculating it for the correct node.